### PR TITLE
Account datasource

### LINKED ISSUE #149: Implement outscale_account datasource(s)

### PR DESCRIPTION
**Pull request by FredericBerot-Armand**
closes #149

### CODE REVIEW

File: outscale/data_source_outscale_account.go [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+			},
+			"request_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceOutscaleOAPIAccountRead(d *schema.ResourceData, meta interface{}) error {
```
Comment (cosmo-ray) [inline outscale/data_source_outscale_account.go:87]: `s/dataSourceOutscaleOAPIAccountRead/dataSourceAcountRead/`
We did use OutscaleOAPI in some function, but this is deprecated
The OAPI tell nothing about the function utility and repeat what `Outscale` already tell
Being insid Outscale Provider, rewrite Outscale in the Function name is also irrelevant. (but more up to debate)

File: outscale/data_source_outscale_account.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+	"time"
+
+	oscgo "github.com/outscale/osc-sdk-go/v2"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-outscale/utils"
+)
+
+func dataSourceOutscaleOAPIAccount() *schema.Resource {
```
Comment (cosmo-ray) [inline outscale/data_source_outscale_account.go:14]: `s/dataSourceOutscaleOAPIAccount/dataSourceAccount`

File: outscale/data_source_outscale_account_test.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package outscale
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func TestAccDataSourceOutscaleOAPIAccount_basic(t *testing.T) {
```
Comment (cosmo-ray) [inline outscale/data_source_outscale_account_test.go:10]: same here, remove `OutscaleOAPI`

File: outscale/data_source_outscale_account_test.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceOutscaleOAPIAccountConfig(),
+			},
+		},
+	})
+}
+
+func testAccDataSourceOutscaleOAPIAccountConfig() string {
```
Comment (cosmo-ray) [inline outscale/data_source_outscale_account_test.go:25]: same here, remove `OutscaleOAPI`

File: outscale/data_source_outscale_accounts.go [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+	"time"
+
+	oscgo "github.com/outscale/osc-sdk-go/v2"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-outscale/utils"
+)
+
+func dataSourceOutscaleOAPIAccounts() *schema.Resource {
```
Comment (cosmo-ray) [inline outscale/data_source_outscale_accounts.go:14]: same here, remove `OutscaleOAPI`

File: outscale/data_source_outscale_accounts.go [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+	return &schema.Resource{
+		Read: dataSourceOutscaleOAPIAccountsRead,
```
Comment (cosmo-ray) [inline outscale/data_source_outscale_accounts.go:16]: same here, remove `OutscaleOAPI`

File: outscale/data_source_outscale_accounts.go [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+			},
+			"request_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceOutscaleOAPIAccountsRead(d *schema.ResourceData, meta interface{}) error {
```
Comment (cosmo-ray) [inline outscale/data_source_outscale_accounts.go:95]: same here, remove `OutscaleOAPI`

File: outscale/data_source_outscale_accounts.go [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+	if err != nil {
+		return err
+	}
+	if err := d.Set("accounts", flattenAccounts(resp.GetAccounts())); err != nil {
+		return err
+	}
+	return nil
+}
+
+func flattenAccounts(accounts []oscgo.Account) []map[string]interface{} {
```
Comment (cosmo-ray) [inline outscale/data_source_outscale_accounts.go:120]: the function name is OK here.

File: outscale/data_source_outscale_accounts_test.go [hunk 9]
```diff
@@ -0,0 +1,29 @@
+package outscale
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func TestAccDataSourceOutscaleOAPIAccounts_basic(t *testing.T) {
```
Comment (cosmo-ray) [inline outscale/data_source_outscale_accounts_test.go:10]: same here, remove `OutscaleOAPI`

File: outscale/data_source_outscale_accounts_test.go [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceOutscaleOAPIAccountsConfig(),
+			},
+		},
+	})
+}
+
+func testAccDataSourceOutscaleOAPIAccountsConfig() string {
```
Comment (cosmo-ray) [inline outscale/data_source_outscale_accounts_test.go:25]: same here, remove `OutscaleOAPI`

File: outscale/provider.go [hunk 11]
```diff
@@ -176,6 +176,8 @@ func Provider() terraform.ResourceProvider {
 			"outscale_api_access_rule":              dataSourceOutscaleOAPIApiAccessRule(),
 			"outscale_api_access_rules":             dataSourceOutscaleOAPIApiAccessRules(),
 			"outscale_api_access_policy":            dataSourceOutscaleOAPIApiAccessPolicy(),
+			"outscale_account":                      dataSourceOutscaleOAPIAccount(),
```
Comment (cosmo-ray) [inline outscale/provider.go:179]: same here, remove `OutscaleOAPI`

**Review by cosmo-ray — CHANGES_REQUESTED** [review]

File: outscale/data_source_outscale_account.go (hunk 1 above)
Comment (FredericBerot-Armand) [reply outscale/data_source_outscale_account.go:87]: I'm ok with that, but actually every function is named like this, for me it's another issues for fix this in every file

Comment (outscale-mgo) [reply outscale/data_source_outscale_account.go:87]: That's partially true, some function have remove the "OAPI". but most have both.
But yes renaming everything is still something we have to do, but a low priority issue.

Comment (FredericBerot-Armand) [reply outscale/data_source_outscale_account.go]: Done

File: outscale/data_source_outscale_account.go [hunk 12]
```diff
@@ -0,0 +1,155 @@
...
+			return utils.CheckThrottling(err)
+		}
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+
+	account := resp.GetAccounts()[0]
+
```
Comment (outscale-toa) [inline outscale/data_source_outscale_account.go:108]: Can you check please if you got more than one element in the read response please?

**Review by outscale-toa — CHANGES_REQUESTED** [review]

**Review by outscale-toa — APPROVED** [review]

**Pull request merged**